### PR TITLE
test(storage): add tests for dual-region buckets

### LINKED ISSUE #5776: storage: Add Support for Custom Dual Region Buckets
**FR:** Support custom dual region buckets.

The method for creating a bucket should accept 2 locations.

**System tests:**
- Verify that a Custom Dual Region bucket can be created
- Verify that a Custom Dual Region bucket’s metadata reflects this when retrieved

[Internal] For details and coordination: [go/gcs-dpe-cdr-milestones](http://go/gcs-dpe-cdr-milestones)

### PR DESCRIPTION
**Pull request by BrennaEpp**
* refactors test to use t.Run 
* adds test for lifecycle rules
* renames the test since it is only testing create and delete, not other bucket methods
* adds the dual-region check

Fixes #5776

### CODE REVIEW

File: storage/integration_test.go [hunk 1]
```diff
@@ -252,103 +252,161 @@ func config(ctx context.Context, scopes ...string) *Client {
...
+				VersioningEnabled: false,
+				LocationType:      "multi-region",
+				Location:          "US",
+			},
+		},
+		{
+			name: "with attrs",
+			attrs: &BucketAttrs{
+				StorageClass:      "NEARLINE",
+				VersioningEnabled: true,
```
Comment (tritone) [inline storage/integration_test.go:334]: Probably makes sense to test using a non-default location for this one.

File: storage/integration_test.go [hunk 2]
```diff
@@ -252,103 +252,161 @@ func config(ctx context.Context, scopes ...string) *Client {
...
-	// Test Create and Delete with attributes.
 	labels := map[string]string{
 		"l1":    "v1",
 		"empty": "",
 	}
-	attrs = &BucketAttrs{
-		StorageClass:      "NEARLINE",
-		VersioningEnabled: true,
-		Labels:            labels,
-		Lifecycle: Lifecycle{
```
Comment (tritone) [inline storage/integration_test.go:293]: Was setting the lifecycle not validated previously? That's not great.

**Review by tritone — APPROVED** [review]
Overall looks good, thanks for the cleanup!

File: storage/integration_test.go (hunk 2 above)
Comment (BrennaEpp) [reply storage/integration_test.go:293]: It was not :/

**Review by danielbankhead — APPROVED** [review]

**Pull request merged**